extension/scripts/features/friendlyfire/ttFriendlyFire.js

```diff
@@ -0,0 +1,40 @@
+"use strict";
+
+(async () => {
+	featureManager.registerFeature(
+		"Friendly Fire",
+		"profile",
+		() => settings.pages.profile.showAllyWarning,
+		null,
+		addWarning,
+		removeWarning,
+		{
+			storage: ["settings.pages.profile.showAllyWarning", "settings.allyFactionsIDs.length"],
+		},
+		null
+	);
+
+	async function addWarning() {
+		if (document.find(".tt-ally-warning")) document.find(".tt-ally-warning").remove();
+		await requireElement(".user-info-value [href*='/factions.php']");
+		let warning;
+		const factionID = parseInt(document.find(".user-info-value [href*='/factions.php']").href.replace(/\D+/g, ""));
+		const factionName = document.find(".user-info-value [href*='/factions.php']").innerText.trim();
+		if (factionID === userdata.faction.faction_id) warning = "This user is in your faction!";
```

Comment:
Make sure that the user has API data for this part.

Reply:
Added.

Reply:
It should be added in this place, not in the general check.